src/net_processing.cpp

```diff
@@ -2936,9 +2936,11 @@ void ProcessMessage(
 
         TxValidationState state;
 
-        nodestate->m_tx_download.m_tx_announced.erase(hash);
-        nodestate->m_tx_download.m_tx_in_flight.erase(hash);
-        EraseTxRequest(hash);
+        for (auto gtxid : {GenTxid{false, txid}, GenTxid{true, wtxid}}) {
+            nodestate->m_tx_download.m_tx_announced.erase(gtxid);
+            nodestate->m_tx_download.m_tx_in_flight.erase(gtxid);
+            EraseTxRequest(gtxid);
```

Comment:
Is this a slight behavior change ?

By erasing both wtxid/txid from `g_already_asked_for`, if we receive a wtxid from a wtixd-relay peer and at same time gets an announcement for the same tx from few txid-relay only peers, when we monitor our getdatas in `SendMessages`, we would send a spurious GETDATA for the first-processed txid-relay peer as `last_request_time` is null.

Reply:
@ariard After a tx arrives successfully, it should be AlreadyHave, and new INVs for it (of whatever type) should be ignored.